### PR TITLE
Fix and improve page leave warnings

### PR DESCRIPTION
**Pull request by loneil**
# Description
Fix so create page asks you if you want to leave again. This is for close tab/browser or refresh.
Add Vue route guards so that you get prompted if you use a nav, go back etc.
Add a dirty flag on change to trigger when to do either of these.

Changes to create new form and edit form.
No changes to Submit mode, since that opens a new instance of the form there's no navigating so keeping it simple as it is.

## Types of changes

Bug fix (non-breaking change which fixes an issue)
New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wxpbtr-tools.pathfinder.gov.bc.ca/dashboard?id=common-hosted-form-service-pr-75

File: app/frontend/src/store/modules/form.js [hunk 1]
```diff
@@ -135,6 +139,14 @@ export default {
     resetForm({ commit }) {
       commit('SET_FORM', genInitialForm());
     },
+    async setDirtyFlag({ commit, state }, isDirty) {
+      // When the form is deteced to be dirty set the browser gaurds for closing the tab etc
```
Comment (parc-jason) [inline app/frontend/src/store/modules/form.js:143]: detected, guards

**Review by parc-jason — APPROVED** [review]
Woot!
That's nice.. I checked on Safari and Chome (mac).

File: app/frontend/src/store/modules/form.js (hunk 1 above)
Comment (loneil) [reply app/frontend/src/store/modules/form.js]: Fixed

**Review by TimCsaky — APPROVED** [review]

**Review by jujaga — APPROVED** [review]

**Pull request merged**